src/model/MultiValueColumn.ts

```diff
@@ -0,0 +1,281 @@
+//
+//       var index = (percentile / 100) * data.length;
+//       var result;
+//       if (Math.floor(index) === index) {
+//         result = (data[(index - 1)] + data[index]) / 2;
+//       } else {
+//         result = data[Math.floor(index)];
+//       }
+//       return result;
+//     }
+
+function numSort(a, b) {
+  return a - b;
+}
+
+
+export default class MultiValueColumn extends ValueColumn<number[] > {
+  private sortBy;
+  private colorrange;
+  private min;
+  private max;
+  private bins;
+  private threshold;
+
+  private ypositionVerticalBar;
+  private verticalBarHeight;
+  private colorScale: d3.scale.Linear<number, string> = d3.scale.linear<number, string>();
+  private xposScale: d3.scale.Linear<number, number> = d3.scale.linear();
+  private yposScale: d3.scale.Linear<number, number> = d3.scale.linear();
+  private verticalBarScale: d3.scale.Linear<number,number> = d3.scale.linear();
+  private boxPlotScale: d3.scale.Linear<number,number> = d3.scale.linear();
+
+
+  constructor(id: string, desc: any) {
+    super(id, desc);
+    this.colorrange = (<any>desc).colorrange || ['blue', 'red'];
+    this.min = d3.min((<any>desc).domain);
+    this.max = d3.max((<any>desc).domain);
+    this.bins = (<any>desc).datalength;
+    this.threshold = (<any>desc).threshold || 0;
+
+    this.sortBy = (<any>desc.sort) || 'min';
+    this.verticalBarHeight = 13;
+    this.rendererList = [{type: 'heatmapcustom', label: 'Heatmap'},
+      {type: 'boxplot', label: 'Boxplot'},
+      {type: 'sparkline', label: 'Sparkline'},
+      {type: 'threshold', label: 'Threshold'},
+      {type: 'verticalbar', label: 'VerticalBar'}];
+
+    this.defineColor();
+    this.boxPlotWidth();
+
+  }
+
+  private defineColor() {
+
+    if (this.min < 0) {
+      this.colorScale
+        .domain([this.min, 0, this.max])
+        .range([this.colorrange[0], 'white', this.colorrange[1]]);
+
+    } else {
+      this.colorScale
+        .domain([this.min, this.max])
+        .range(['white', this.colorrange[1]]);
+    }
+  }
+
+
+  private boxPlotWidth() {
+    this.boxPlotScale
+      .domain([this.min, this.max])
+      .range([0, this.getWidth()])
+
+  }
+
+
+  compare(a: any, b: any, aIndex: number, bIndex: number) {
+    this.sortBy = this.getUserSortBy();
+    const a_val = this.getValue(a, aIndex);
+    const b_val = this.getValue(b, bIndex);
+    const sort: any = new CustomSortCalculation(a_val, b_val);
+    const f = sort[this.sortBy].bind(sort);
+
+    return f();
+  }
+
+
+  getLabel(row: any, index: number) {
+    return '' + this.getValue(row, index);
+  }
+
+  getRaw(row: any, index: number) {
+    return this.accessor(row, index, this.id, this.desc, this.findMyRanker());
+  }
+
+  getValue(row: any, index: number) {
+    var v = this.getRaw(row, index);
+    return (v);
+  }
+
+  getColor(data: any) {
+
+    return this.colorScale(data);
+  }
+
+  calculateCellDimension() {
+
+    return (this.getWidth() / this.bins);
+  }
+
+  getxScale(data) {
```

Comment:
In general, default values for different properties and scaling to a common scale (e.g. 0.1) ... part of model. Everything renderer specific (like mapping to width, computing positions) -> renderer